apps/sync/test/unit/crypto/stringconversion_test.js

```diff
@@ -85,6 +99,22 @@ suite('utils', () => {
           throw(Error);
     });
   });
+  suite('byteArrayToUtf16String', () => {
+    test('converts an Uint8Array to a UTF-16 string', () => {
+      var array = new Uint8Array(3);
+      array[0] = 0xe2;
+      array[1] = 0x82;
+      array[2] = 0xac;
```

Comment:
That's no doubt that [0xe2, 0x82,0xac] is an byte array in UTF-8 code.
I mentioned [here](https://github.com/mozilla-b2g/gaia/pull/31903/files#r41014832) that "byteArrayToUtf16String" is converting UTF8 byte array to UTF-16. 
So '€' in UTF-16 has to be verified at [here](https://github.com/mozilla-b2g/gaia/pull/31903/files#diff-5a3d47b065aef47f4604389e7e708062R110)

I think this issue is still related to the converter of UTF-8 <-> UTF-16.